### ISSUE TITLE
`@angular/cli@14.2` `ng new` command installing Angular `14.0.0-next.0` packages

### ISSUE THREAD

**Issue by MitchellCash** [angular/angular-cli#23804]
# 🐞 Bug report

### Command (mark with an `x`)

- [x] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

Yes, this issue was not present in Angular 14.1.3.

### Description

With Angular `@angular/cli@14.2` when running `ng new [name]` it will install Angular `14.0.0-next.0` packages. See below:

```json
  "dependencies": {
    "@angular/animations": "^14.0.0-next.0",
    "@angular/common": "^14.0.0-next.0",
    "@angular/compiler": "^14.0.0-next.0",
    "@angular/core": "^14.0.0-next.0",
    "@angular/forms": "^14.0.0-next.0",
    "@angular/platform-browser": "^14.0.0-next.0",
    "@angular/platform-browser-dynamic": "^14.0.0-next.0",
    "@angular/router": "^14.0.0-next.0",
    "rxjs": "~7.5.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.11.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^14.2.0",
    "@angular/cli": "~14.2.0",
    "@angular/compiler-cli": "^14.0.0-next.0",
    "@types/jasmine": "~4.0.0",
    "jasmine-core": "~4.3.0",
    "karma": "~6.4.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage": "~2.2.0",
    "karma-jasmine": "~5.1.0",
    "karma-jasmine-html-reporter": "~2.0.0",
    "typescript": "~4.7.2"
  }
```

## 🔬 Minimal Reproduction

1. Install `@angular/cli@14.2`
2. Run `ng new example`
3. Take note of the `package.json` package versions

## 🔥 Exception or Error

N/A

## 🌍 Your Environment

```
Angular CLI: 14.2.0
Node: 16.17.0
Package Manager: npm 8.15.0 
OS: darwin x64

Angular: 14.2.0
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1402.0
@angular-devkit/build-angular   14.2.0
@angular-devkit/core            14.2.0
@angular-devkit/schematics      14.2.0
@schematics/angular             14.2.0
rxjs                            7.5.6
typescript                      4.7.4
```

**Anything else relevant?**

N/A

**Comment by MitchellCash** [comment]
I should note that because we are using a caret version range (i.e. `^14.0.0-next.0`), that we do end up installing `14.2.0` regardless. However, as per previous versions of `@angular/cli`, the minimum version of `^14.2.0` should be used to avoid accidental incompatibilities.

**Comment by dgp1130** [comment]
Just pushed out a fix in 14.2.1. Make sure your CLI is up to date, and it should generate projects with stable versions.
